components/brave_wallet_ui/page/screens/onboarding/restore-from-recovery-phrase/restore-from-extension.tsx

```diff
@@ -0,0 +1,318 @@
+
+type RestoreFromExtensionSteps = 'newPassword' | 'currentPassword'
+
+interface Props {
+  restoreFrom: 'metamask' | 'legacy'
+}
+
+export const OnboardingRestoreFromExtension = ({ restoreFrom }: Props) => {
+  // routing
+  let history = useHistory()
+
+  // queries
+  const { isFetching: isCheckingExtensions } = useGetWalletsToImportQuery()
+
+  // mutations
+  const [report] = useReportOnboardingActionMutation()
+
+  const [checkExtensionPassword, { isLoading: isCheckingImportPassword }] =
+    useCheckExternalWalletPasswordMutation()
+
+  const [
+    importFromCryptoWallets,
+    {
+      data: importFromLegacyWalletResult,
+      isLoading: isImportingFromLegacyExtension
+    }
+  ] = useImportFromCryptoWalletsMutation()
+
+  const [
+    importFromMetaMask,
+    {
+      data: importFromMetaMaskResult,
+      isLoading: isImportingFromMetaMaskExtension
+    }
+  ] = useImportFromMetaMaskMutation()
+
+  // computed from mutations
+  const isCreatingWallet =
+    (restoreFrom === 'legacy' && isImportingFromLegacyExtension) ||
+    (restoreFrom === 'metamask' && isImportingFromMetaMaskExtension)
+
+  // state
+  const [isPasswordValid, setIsPasswordValid] = React.useState(false)
+  const [password, setPassword] = React.useState('')
+  const [extensionPassword, setExtensionPassword] = React.useState('')
+  const [extensionPasswordError, setExtensionPasswordError] = React.useState('')
+  const [currentStep, setCurrentStep] =
+    React.useState<RestoreFromExtensionSteps>('currentPassword')
+
+  const importWalletError =
+    extensionPasswordError ||
+    (restoreFrom === 'metamask' && importFromMetaMaskResult?.errorMessage) ||
+    (restoreFrom === 'legacy' && importFromLegacyWalletResult?.errorMessage)
+
+  // methods
+  const checkImportPassword = React.useCallback(async () => {
+    const results = await checkExtensionPassword({
+      walletType:
+        restoreFrom === 'metamask'
+          ? BraveWallet.ExternalWalletType.MetaMask
+          : BraveWallet.ExternalWalletType.CryptoWallets,
+      password: extensionPassword
+    }).unwrap()
+    if (results.errorMessage) {
+      setExtensionPasswordError(results.errorMessage)
+    } else {
+      setExtensionPasswordError('')
+      setCurrentStep('newPassword')
+    }
+  }, [extensionPassword])
+
+  const restoreWallet = React.useCallback(async () => {
+    if (!isPasswordValid) {
+      return
+    }
+
+    if (restoreFrom === 'metamask') {
+      await importFromMetaMask({
+        password: extensionPassword,
+        newPassword: password
+      }).unwrap()
+      history.push(WalletRoutes.OnboardingComplete)
+      return
+    }
+
+    if (restoreFrom === 'legacy') {
+      await importFromCryptoWallets({
+        password: extensionPassword,
+        newPassword: password
+      }).unwrap()
+      history.push(WalletRoutes.OnboardingComplete)
+    }
+  }, [isPasswordValid, restoreFrom, extensionPassword, password])
```

Comment:
Missing `importFromMetaMask` and `importFromCryptoWallets` dependencies.

Reply:
Thanks, fixed